fosslight/fosslight
# Apply Checkstyle for checking Java code quality

## Description
I added [Checkstyle](https://checkstyle.sourceforge.io) settings with [Google Style Guide](https://google.github.io/styleguide/javaguide.html).

### Checkstyle

[Checkstyle](https://checkstyle.sourceforge.io) is a development tool to help programmers write Java code that adheres to a coding standard.  This tool can be used in [IDE plugins](https://plugins.jetbrains.com/plugin/1065-checkstyle-idea) or via Maven and Gradle.

**Use Gradle**

```
./gradlew -q checkStyleMain
./gradlew -q checkStyleTest
```

**Or use [IntelliJ plugin](https://plugins.jetbrains.com/plugin/1065-checkstyle-idea)**

Step 1: Download the Checkstyle plugin.
<img width="1034" alt="plugin" src="https://user-images.githubusercontent.com/44025432/180656242-a8e6de5a-7da4-43d1-8799-a5e7e66d66b9.png">
Step 2: Configure the Checkstyle plugin to use the style guide file
1. In the settings window, search for "checkstyle" in the left-hand nav search bar and select _Checkstyle_.
<img width="1034" alt="search check style" src="https://user-images.githubusercontent.com/44025432/180656313-f1461d7b-3d24-48c5-8279-f3275a259a7f.png">

2. Click the “+” button on the right-hand side of the settings window.

3. Enter a description for the checks.
<p align="center">
  <img width="655" alt="enter a description" src="https://user-images.githubusercontent.com/44025432/180656360-0a172ee8-903d-43b2-99bf-9d84bfad55a2.png">
</p>

4. Select “Use a local Checkstyle file”.

5. Select `config/checkstyle/checkstyle.xml`.

6. Once the style file has been added, select the check box to the left to make  newly created guide the default style checker.

7. Click OK and you’re done.

### Google Java Code Formatter in IntelliJ IDEA
<p align="center">
  <img width="512" alt="formatter" src="https://user-images.githubusercontent.com/44025432/180656512-91a1729c-e0b3-44df-aac7-23fb93076654.gif">
</p>

**How to use**
- Reformat Code
  - Windows: `Ctrl` + `Alt` + `L`
  - macOS: `Opt` + `Cmd` + `L`

- Optimize Imports
  - Windows: `Ctrl` + `Alt` + `O`
  - macOS: `Ctrl` + `Opt` + `O`

**How to configure Google Java Code Formatter**
Step 1: In the settings window, search for "java" in the left-hand nav search bar and select `Editor > Code Style > Java`.
<img width="1034" alt="search code style setting" src="https://user-images.githubusercontent.com/44025432/180656961-eade5668-7d2a-4cff-9787-de1648c24fa0.png">

Step 2: Next to the Scheme drop-down menu select the icon then Import Scheme > IntelliJ IDEA code style XML then select the `config/intellij-java-google-style.xml`.

## Additional context
As a result of running the Checkstyle, I think a lot of code changes will occur. 

## Type of change

- [ ] Refactoring, Maintenance
